pkg/kn/commands/service/update_test.go

```diff
@@ -726,6 +727,14 @@ func TestServiceUpdateNoClusterLocalOnPrivateService(t *testing.T) {
 	assert.DeepEqual(t, expected, actual)
 }
 
+func TestServiceUpdateDeletionTimestampNotNil(t *testing.T) {
+	original := newEmptyService()
+	original.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+	_, _, _, err := fakeServiceUpdate(original, []string{
+		"service", "update", "foo", "--revision-name", "foo-v1"})
+	assert.ErrorContains(t, err, fmt.Sprintf("the service %s has been deleted", original.Name))
```

Comment:
```suggestion
	assert.ErrorContains(t, err, original.Name))
	assert.ErrorContains(t, err, "deleted"))
	assert.ErrorContains(t, err, "service"))
```

Just check for keywords to make the test more robust (e.g. give it a chance to not fail if the wording of the error is changed but the essential context is still contained). That's the whole purpose to use `Contains` over `Equals`)